src/components/RepositoryNavigation.js

```diff
@@ -0,0 +1,177 @@
+import ListGroupItem from 'react-bootstrap/ListGroupItem';
+import { LinkContainer } from 'react-router-bootstrap';
+import { getContents } from '../util/GithubRepositoryUtil';
+import { REPO_RESOURCE, RENDER_RESOURCE } from '../util/Constants';
+import queryString from 'query-string';
+
+/**
+ * A React component for rendering repository items when navigating the /repos resource
+ */
+class RepositoryNavigation extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      repository: {
+        isLoaded: false,
+        contents: []
+      }
+    };
+  }
+
+  /**
+   * When component first initializes, execute following lifecycle actions
+   */
+  async componentDidMount() {
+    const queryStringMap = queryString.parse(this.props.location.search);
+    const owner = queryStringMap.owner;
+    const repo = queryStringMap.repo;
+    const path = queryStringMap.path;
+
+    const json = await getContents(owner, repo, path);
+    this.setState({
+      repository: {
+        isLoaded: true,
+        queryStringMap: queryStringMap,
+        contents: json
+      }
+    });
+  }
+
+  /**
+   * When a page is reloaded with a new query string, then following lifecycle actions
+   * @param {Object} newProps Updated query string
+   */
+  async componentWillReceiveProps(newProps) {
+    const newQueryStringMap = queryString.parse(newProps.location.search);
+    const owner = this.state.repository.queryStringMap.owner;
+    const repo = this.state.repository.queryStringMap.repo;
+    const path = newQueryStringMap.path;
+
+    if (newQueryStringMap !== this.state.repository.queryStringMap) {
+      const json = await getContents(owner, repo, path);
+      this.setState({
+        repository: {
+          isLoaded: true,
+          queryStringMap: newQueryStringMap,
+          contents: json
+        }
+      });
+    }
+  }
+
+  render() {
+    const { isLoaded } = this.state.repository;
+    if (!isLoaded) {
+      return (
+        <div className="Markdown">Loading...</div>
+      );
+    } else {
+      const listGroupItems = [];
+      const contents = this.state.repository.contents;
+      const queryStringMap = this.state.repository.queryStringMap;
+
+      contents.forEach((item) => {
+        // Create a new copy of the old query string map, then modify it with references to each
+        // item's relative path
+        const resourceQueryStringCopy = { ...queryStringMap };
+        const key = `list-group-item-${item.name}`;
+        resourceQueryStringCopy.path = `/${item.path}`;
+        const newQueryString = queryString.stringify(resourceQueryStringCopy);
+        listGroupItems.push(
+          item.type === 'dir' ? (
+            <LinkContainer to={`${REPO_RESOURCE}?${newQueryString}`} key={key}>
+              <ListGroupItem action>
+                <i>{`/${item.name}`}</i>
+              </ListGroupItem>
+            </LinkContainer>
+          ) : item.type === 'file' ? (
```

Comment:
IMO nested ternary operators should be split out to `if/else if/else` statements. Hard to read exactly what's going on here.